bridge/engine/EngineMixer.cpp

```diff
@@ -3834,13 +3845,48 @@ void EngineMixer::checkIfRateControlIsNeeded(const uint64_t timestamp)
     _probingVideoStreams = enableBEProbing;
 }
 
-void EngineMixer::runTransportTicks(const uint64_t timestamp)
+template <typename TStream>
+void removeIdleStreams(concurrency::MpmcHashmap32<size_t, TStream*>& streams,
+    EngineMixer* mixer,
+    const uint64_t timestamp)
 {
-    for (auto videoIt : _engineVideoStreams)
+    for (const auto& it : streams)
     {
-        auto& videoStream = *videoIt.second;
-        videoStream.transport.runTick(timestamp);
+        auto stream = it.second;
+        if (!stream->idleTimeoutSeconds)
+        {
+            continue;
+        }
+
+        if (utils::Time::diffLT(stream->createdAt, timestamp, stream->idleTimeoutSeconds * utils::Time::sec))
+        {
+            continue;
+        }
+
+        const auto lastReceivedTimestamp = stream->transport.getLastReceivedPacketTimestamp();
+        if (utils::Time::diffGE(lastReceivedTimestamp, timestamp, stream->idleTimeoutSeconds * utils::Time::sec))
+        {
+            logger::info("Removing idle stream for endpoint %zu, last received packet timestamp: %zu, current: %zu",
+                "EngineMixer",
+                stream->endpointIdHash,
+                lastReceivedTimestamp,
+                timestamp);
+            mixer->removeStream(stream);
+        }
+    }
+}
+
+void EngineMixer::removeIdleStreams(const uint64_t timestamp)
+{
+    if (utils::Time::diffLT(_lastIdleTransportCheck, timestamp, 1ULL * utils::Time::sec))
+    {
+        return;
     }
+    _lastIdleTransportCheck = timestamp;
+
+    ::bridge::removeIdleStreams<EngineVideoStream>(_engineVideoStreams, this, timestamp);
```

Comment:
It looks like if you put template param on container type instead, you do not need to specify the type at all here. It will automagically resolve in the removeIdleStreams method.